test/storage_wipe_test.go

```diff
@@ -0,0 +1,272 @@
+							containerName, false, cloudProvider)
+						aeroCluster := getStorageInitAerospikeCluster(
+							clusterNamespacedName, *storageConfig, racks, latestImage)
+
+						aeroCluster.Spec.PodSpec = podSpec
+
+						By("Cleaning up previous pvc")
+
+						err := cleanupPVC(k8sClient, namespace)
+						Expect(err).ToNot(HaveOccurred())
+
+						By("Deploying the cluster")
+
+						err = deployCluster(k8sClient, ctx, aeroCluster)
+						Expect(err).ToNot(HaveOccurred())
+
+						By("Writing some data to the all volumes")
+						err = writeDataToVolumes(aeroCluster)
+						Expect(err).ToNot(HaveOccurred())
+
+						err = checkIfVolumesWiped(aeroCluster)
+						if err != nil {
+							switch err.(type) {
+							case UnwipedVolumesError:
+								Expect(err).To(HaveOccurred(), "volumes are not wiped")
+							default:
+								Expect(err).ToNot(HaveOccurred())
+
+							}
+						} else {
+							Expect(nil).ToNot(HaveOccurred(), "volumes should not be zeroed")
+						}
+
+						By(fmt.Sprintf("Downgrading image from %s to %s - volumes should not be wiped",
+							latestImage, version6))
+						err = UpdateClusterImage(aeroCluster, version6Image)
+						Expect(err).ToNot(HaveOccurred())
+						err = aerospikeClusterCreateUpdate(
+							k8sClient, aeroCluster, ctx,
+						)
+						Expect(err).ToNot(HaveOccurred())
+
+						By("Checking if volumes are wiped")
+						err = checkIfVolumesWiped(aeroCluster)
+						if err != nil {
+							switch err.(type) {
+							case UnwipedVolumesError:
+								Expect(err).To(HaveOccurred(), "volumes are not wiped")
+							default:
+								Expect(err).ToNot(HaveOccurred())
+
+							}
+						} else {
+							Expect(nil).ToNot(HaveOccurred(), "volumes should not be wiped")
+						}
+
+						By(fmt.Sprintf("Downgrading image from %s to %s - volumes should be wiped",
+							version6, prevServerVersion))
+						err = UpdateClusterImage(aeroCluster, prevImage)
+						Expect(err).ToNot(HaveOccurred())
+						err = aerospikeClusterCreateUpdate(
+							k8sClient, aeroCluster, ctx,
+						)
+						Expect(err).ToNot(HaveOccurred())
+
+						By("Checking if volumes are wiped")
+						err = checkIfVolumesWiped(aeroCluster)
+						Expect(err).ToNot(HaveOccurred())
+
+						if aeroCluster != nil {
+
+							By("Updating the volumes to cascade delete so that volumes are cleaned up")
+							aeroCluster.Spec.Storage.BlockVolumePolicy.InputCascadeDelete = &trueVar
+							aeroCluster.Spec.Storage.FileSystemVolumePolicy.InputCascadeDelete = &trueVar
+							err := aerospikeClusterCreateUpdate(
+								k8sClient, aeroCluster, ctx,
+							)
+							Expect(err).ToNot(HaveOccurred())
+
+							err = deleteCluster(k8sClient, ctx, aeroCluster)
+							Expect(err).ToNot(HaveOccurred())
+
+							pvcs, err := getAeroClusterPVCList(
+								aeroCluster, k8sClient,
+							)
+							Expect(err).ToNot(HaveOccurred())
+
+							Expect(len(pvcs)).Should(
+								Equal(0), "PVCs not deleted",
+							)
+						}
+					},
+				)
+			},
```

Comment:
You can add one more test where a non-aerospike volume is added. Meaning a volume is created but is not used for Aerospike. This block volume should not be wiped after an upgrade.

Reply:
Those volumes will not be used in the namespace storage.